### ISSUE TITLE
External functions somehow require storage pointers

### ISSUE THREAD

**Issue by chriseth** [ethereum/solidity#4771]
```
contract c {
    function f(bytes calldata ) external {}
    function g() public {
        bytes memory a;
        this.f(a);
    }
}
```
results in 
```
browser/Untitled12.sol:5:16: TypeError: Invalid type for argument in function call. Invalid implicit conversion from bytes memory to bytes storage pointer requested.
        this.f(a);
               ^
```

**Comment by chriseth** [comment]
This might be fixed by the data location PR.

**Comment by chriseth** [comment]
It only happens if there is an explicit `calldata` keyword, that's probably why we did not have tests (yet).

**Comment by chriseth** [comment]
Fixed in #4390  but will be fixed again in #4738
